### PR TITLE
closeAttemptExceptionWillBeThrownDownTheCallStack test is added

### PR DESCRIPTION
**Pull request by terziele**
…ing AutoClosable ExtensionContext(that's all classes that extends  AbstractExtensionContext), this exception will be thrown further up the call stack.

## Overview

Hi everyone. Just started to investigate the JUnit5 source code and decided to contribute a bit to this awesome project by covering with tests some uncovered code lines. 

---

I hereby agree to the terms of the [JUnit Contributor License Agreement](https://github.com/junit-team/junit5/blob/002a0052926ddee57cf90580fa49bc37e5a72427/CONTRIBUTING.md#junit-contributor-license-agreement).

---

### Definition of Done

- [x] There are no TODOs left in the code
- [ ] Method [preconditions](http://junit.org/junit5/docs/snapshot/api/org/junit/platform/commons/util/Preconditions.html) are checked and documented in the method's Javadoc
- [ ] [Coding conventions](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#coding-conventions) (e.g. for logging) have been followed
- [ ] Change is covered by [automated tests](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#tests) including corner cases, errors, and exception handling
- [ ] Public API has [Javadoc](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#javadoc) and [`@API` annotations](http://junit.org/junit5/docs/snapshot/api/org/junit/platform/commons/meta/API.html)
- [ ] Change is documented in the [User Guide](http://junit.org/junit5/docs/snapshot/user-guide/) and [Release Notes](http://junit.org/junit5/docs/snapshot/user-guide/#release-notes)
- [ ] All [continuous integration builds](https://github.com/junit-team/junit5#continuous-integration-builds) pass

### CODE REVIEW

File: junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/execution/JupiterEngineExecutionContextTests.java [hunk 1]
```diff
@@ -86,4 +89,20 @@ void canOverrideAttributeWhenContextIsExtended() {
 		assertSame(testInstanceProvider, newContext.getTestInstanceProvider());
 	}
 
+	@Test
+	void closeAttemptExceptionWillBeThrownDownTheCallStack() {
+		FailingAutoClosableExtenstionContext failingExtensionContext = mock(FailingAutoClosableExtenstionContext.class);
+		doCallRealMethod().when(failingExtensionContext).close();
```
Comment (marcphilipp) [inline junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/execution/JupiterEngineExecutionContextTests.java]: Why use a mock and a real class? I'd prefer just the mock with a stubbed exception in this case.

**Comment by terziele** [comment]
Hm... AppVeyor build seems to be failed because of some network issues.

File: junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/execution/JupiterEngineExecutionContextTests.java [hunk 2]
```diff
@@ -86,4 +90,15 @@ void canOverrideAttributeWhenContextIsExtended() {
...
 
+	@Test
+	void closeAttemptExceptionWillBeThrownDownTheCallStack() throws Exception {
+		ExtensionContext failingExtensionContext = mock(ExtensionContext.class,
+			withSettings().extraInterfaces(AutoCloseable.class));
+		doThrow(Exception.class).when(((AutoCloseable) failingExtensionContext)).close();
+		JupiterEngineExecutionContext newContext = originalContext.extend().withExtensionContext(
+			failingExtensionContext).build();
+
+		assertThrows(Exception.class, newContext::close);
```
Comment (marcphilipp) [inline junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/execution/JupiterEngineExecutionContextTests.java]: I think we should check that the thrown exception is the same as the one stubbed above.

Comment (terziele) [reply junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/execution/JupiterEngineExecutionContextTests.java]: Done.

**Comment by marcphilipp** [comment]
I manually merged this to `master` in 51f2b6ceabe330faf05256079ceea9be72a22cc1 and polished it a bit in 2cfad49b409011038acbbd5b795be4acfe6cd732.

Thanks, @terziele! 👍

**Comment by terziele** [comment]
Oh, so that's the way to check the logger! Thanks, @marcphilipp !
